UPGRADING.md

```diff
@@ -1,6 +1,131 @@
+
+The following properties/methods have been removed from the `Bugsnag` client:
+
+| v5.x API                                                           | v6.x API                                                          |
+| ------------------------------------------------------------------ | ----------------------------------------------------------------- |
+| `Bugsnag.LeaveBreadcrumb(message)`                                 | Replaced by `Bugsnag.LeaveBreadcrumb(message, metadata, type)`    |
+| `Bugsnag.LeaveBreadcrumb(message, type, metadata)`                 | Replaced by `Bugsnag.LeaveBreadcrumb(message, metadata, type)`    |
+| `Bugsnag.LeaveBreadcrumb(breadcrumb)`                              | Replaced by `Bugsnag.LeaveBreadcrumb(message, metadata, type)`    |
+
+
+### Callbacks
+
+#### Additions
+
+The following methods have been added to the `Bugsnag` class: 
+
+| Property/Method                                                    | Usage                                                             |
+| ------------------------------------------------------------------ | ----------------------------------------------------------------- |
+| `AddOnError` | Add a callback to modify or discard a event before it is recorded.  LINK_DOCS   | 
+|  `AddOnSendError`  | Add a callback to modify or discard a event before it is sent (including native Cocoa and Android events).  LINK_DOCS   | 
+| `AddOnSession` | Add a callbacks to modify or discard sessions before they are recorded and sent.  LINK_DOCS   | 
+
+ 
+#### Deprecations
+
+The following property has been removed from the `Bugsnag` client:
+
+| v5.x API                                                           | v6.x API                                                          |
+| ------------------------------------------------------------------ | ----------------------------------------------------------------- |
+| `Bugsnag.BeforeNotify`                                             | Replaced by `Bugsnag.AddOnSendError(callback) or Bugsnag.AddOnError(callback)`|
+
+### Misc
+
+#### Deprecations
+
+The following methods have been removed from the `Bugsnag` client:
+
+| v5.x API                                                           | v6.x API                                                          |
+| ------------------------------------------------------------------ | ----------------------------------------------------------------- |
+| `Bugsnag.SetAutoNotify(bool)`                                      | Deprecated - no longer public API.  Callbacks should now be used to stop sending particular errors at runtime|
+| `Bugsnag.SetAutoDetectErrors(bool)`                                      | Deprecated - no longer public API.  Callbacks should now be used to stop sending particular errors at runtime|
+| `Bugsnag.SetAutoDetectAnrs(bool)`                                  | Deprecated - no longer public API.  Callbacks should now be used to stop sending particular errors at runtime|
```

Comment:
We should say these are replaced by the `AutoDetectErrors` and `EnabledErrorTypes` configuration options or controlled in an on-send callback.

Reply:
Done 👍